src/main/java/uk/nhs/digital/nhsconnect/nhais/service/SequenceService.java

```diff
@@ -37,6 +39,14 @@ private void validateSenderAndRecipient(String sender, String recipient) {
         }
     }
 
+    private void validateSender(String sender) {
+        if (StringUtils.isBlank(sender)) {
+            throw new SenderValidationException(
+                String.format("Sender not valid. Sender: %s", sender)
+            );
+        }
```

Comment:
This message will be one of this two:
`"Sender not valid. Sender: "` (for blank string)
or
`"Sender not valid. Sender: null"` (for null string)
which won't be really helpful/meaningful.

Also exception could be named `SenderMissingException` to be more precise, take `sender` string as param in constructor and deal explicitly with those two options "inside"

Reply:
It is the same for SenderOrRecipientMissingError - athough I will try make it more readable

Reply:
I always try to replace "valid" with different word(s) ("How to name this without using valid/validate?"). "Valid" is strongly context-related and might mean anything